lib/nrf_modem_lib/lte_connectivity/lte_connectivity.c

```diff
@@ -92,6 +199,48 @@ static void connection_timeout_schedule(void)
 	}
 }
 
+/* Work handler that marks the serving cell as lost after a timeout. */
+static void reg_timeout_work_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	/* Mark the serving cell as lost and update activity accordingly. */
+	update_has_cell(false);
+}
+
+static void reg_timeout_schedule(void)
+{
+	int timeout_seconds;
+	int size_out = sizeof(timeout_seconds);
+	bool has_cell;
+
+	(void)k_mutex_lock(&internal_state_lock, K_FOREVER);
+
+	has_cell = internal_state.has_cell;
```

Comment:
But in this case, you are not checking several at once, only a single bool.

Reply:
So you are suggesting that I use both `atomic_t` and the mutex, and rely solely on `atomic_t` when performing reads?

I can modify the PR to do that if you think that would be cleaner. I'll add you as a reviewer on that PR as well

Reply:
I put together a temporary commit to show what this would look like:
https://github.com/nrfconnect/sdk-nrf/commit/d59260bfe73a9e2762f9aca134d445c555199e25

IMO all the extra `atomic_get` calls outweigh the loss of some of the `k_mutex_lock/unlock` calls.

Furthermore, the only lock/unlock calls that get removed are in locations I plan to replace with a single API call as part of `IRIS-6363`.

The location you specifically commented on here actually doesn't even exist anymore in the CEREG PR

Reply:
I'm aware this is no longer a good place to comment.  Just wanted to share this, from the docs about atomic_t:
`Using atomic variables is typically far more efficient than using other techniques to implement critical sections such as using a mutex or locking interrupts.`

Reply:
> I'm aware this is no longer a good place to comment. 

That's not what I mean,t sorry. I just meant, the perceived benefit no longer exists

> Using atomic variables is typically far more efficient than using other techniques to implement critical sections such as using a mutex or locking interrupts.

It's good advice -- I use atomic_t wherever I can! In this case I chose to fully rely on the mutex because I am forced to use one either way